### ISSUE TITLE
Migrations that rename table try to remove fks from the new table name before renaming the table

### ISSUE THREAD

**Issue by stephaniesyed** [payloadcms/payload#4673]
### Describe the Bug

I am trying to rename entities (User to Employee, ShipperLocation to Partner Site) due to marketing finalizing terms. The migration file generated tries to remove a foreign key from the new table name before renaming the table. 

Console Output to generate the migration file and attempt to migrate: 
```
stephanie.syed@LXYWG0WJ6C partner-manager % make migration MESSAGE="rename shipperLocation and user"
[ -n "rename shipperLocation and user" ] || { echo "MESSAGE=\"descriptive_title\" required"; exit 1; }
Running in docker: yarn run payload migrate:create 'rename shipperLocation and user'
yarn run v1.22.19
$ cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload migrate:create 'rename shipperLocation and user'
[19:34:08] INFO (payload): Starting Payload...
~ users_roles › employees_roles table will be renamed
~ users › employees table will be renamed
~ users_rels › employees_rels table will be renamed
~ shipper_locations › partner_sites table will be renamed
~ shipper_locations_rels › partner_sites_rels table will be renamed
--- all table conflicts resolved ---


~ shipper_locations_id › partner_sites_id column will be renamed
--- all columns conflicts in portals_rels table resolved ---


~ users_id › employees_id column will be renamed
--- all columns conflicts in payload_preferences_rels table resolved ---

~ employees_roles › users_roles table will be renamed
~ employees › users table will be renamed
~ employees_rels › users_rels table will be renamed
~ partner_sites › shipper_locations table will be renamed
~ partner_sites_rels › shipper_locations_rels table will be renamed
--- all table conflicts resolved ---


~ partner_sites_id › shipper_locations_id column will be renamed
--- all columns conflicts in portals_rels table resolved ---


~ employees_id › users_id column will be renamed
--- all columns conflicts in payload_preferences_rels table resolved ---

[19:34:51] INFO (payload): Migration created at /home/node/app/src/migrations/20240103_193408_rename_shipperLocation_and_user.ts
[19:34:51] INFO (payload): Done.
Done in 44.30s.
stephanie.syed@LXYWG0WJ6C partner-manager % make migrate
Running in docker: yarn run payload migrate
yarn run v1.22.19
$ cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload migrate
[19:37:01] INFO (payload): Starting Payload...
[19:37:01] INFO (payload): Reading migration files from /home/node/app/src/migrations
[19:37:01] INFO (payload): Migrating: 20240103_193408_rename_shipperLocation_and_user
[19:37:01] ERROR (payload): Error running migration 20240103_193408_rename_shipperLocation_and_user relation "employees_roles" does not exist.
    err: {
      "type": "DatabaseError",
      "message": "relation \"employees_roles\" does not exist",
      "stack":
          error: relation "employees_roles" does not exist
              at /home/node/app/node_modules/pg-pool/index.js:45:11
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      "length": 113,
      "name": "error",
      "severity": "ERROR",
      "code": "42P01",
      "file": "namespace.c",
      "line": "426",
      "routine": "RangeVarGetRelidExtended"
    }
[19:37:01] INFO (payload): Done.
Done in 1.60s.
```

The migration file references the new table name when trying to drop fks in the upgrade function, but since the table hasn't been renamed yet it causes an error.

Relevant lines from the migration file: 
```
xport async function up({ payload }: MigrateUpArgs): Promise<void> {
await payload.db.drizzle.execute(sql`
...
ALTER TABLE "employees_roles" DROP CONSTRAINT "users_roles_parent_id_users_id_fk";

ALTER TABLE "employees_rels" DROP CONSTRAINT "users_rels_parent_id_users_id_fk";

ALTER TABLE "employees_rels" DROP CONSTRAINT "users_rels_partners_id_partners_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_parent_id_shipper_locations_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_partners_id_partners_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_deliverable_postal_codes_id_deliverable_postal_codes_id_fk";
...
ALTER TABLE "users_roles" RENAME TO "employees_roles";
ALTER TABLE "users" RENAME TO "employees";
ALTER TABLE "users_rels" RENAME TO "employees_rels";
ALTER TABLE "shipper_locations" RENAME TO "partner_sites";
ALTER TABLE "shipper_locations_rels" RENAME TO "partner_sites_rels";
```

Full migration file:
```
import { MigrateUpArgs, MigrateDownArgs } from '@payloadcms/db-postgres'
import { sql } from 'drizzle-orm'

export async function up({ payload }: MigrateUpArgs): Promise<void> {
await payload.db.drizzle.execute(sql`

DO $$ BEGIN
 CREATE TYPE "enum_employees_roles" AS ENUM('super-admin', 'admin');
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

ALTER TABLE "portals_rels" DROP CONSTRAINT "portals_rels_shipper_locations_id_shipper_locations_id_fk";

ALTER TABLE "payload_preferences_rels" DROP CONSTRAINT "payload_preferences_rels_users_id_users_id_fk";

ALTER TABLE "employees_roles" DROP CONSTRAINT "users_roles_parent_id_users_id_fk";

ALTER TABLE "employees_rels" DROP CONSTRAINT "users_rels_parent_id_users_id_fk";

ALTER TABLE "employees_rels" DROP CONSTRAINT "users_rels_partners_id_partners_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_parent_id_shipper_locations_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_partners_id_partners_id_fk";

ALTER TABLE "partner_sites_rels" DROP CONSTRAINT "shipper_locations_rels_deliverable_postal_codes_id_deliverable_postal_codes_id_fk";

ALTER TABLE "users_roles" RENAME TO "employees_roles";
ALTER TABLE "users" RENAME TO "employees";
ALTER TABLE "users_rels" RENAME TO "employees_rels";
ALTER TABLE "shipper_locations" RENAME TO "partner_sites";
ALTER TABLE "shipper_locations_rels" RENAME TO "partner_sites_rels";
ALTER TABLE "portals_rels" RENAME COLUMN "shipper_locations_id" TO "partner_sites_id";
ALTER TABLE "payload_preferences_rels" RENAME COLUMN "users_id" TO "employees_id";
ALTER TABLE "employees_roles" ALTER COLUMN "value" SET DATA TYPE enum_employees_roles;
DO $$ BEGIN
 ALTER TABLE "portals_rels" ADD CONSTRAINT "portals_rels_partner_sites_id_partner_sites_id_fk" FOREIGN KEY ("partner_sites_id") REFERENCES "partner_sites"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "payload_preferences_rels" ADD CONSTRAINT "payload_preferences_rels_employees_id_employees_id_fk" FOREIGN KEY ("employees_id") REFERENCES "employees"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "employees_roles" ADD CONSTRAINT "employees_roles_parent_id_employees_id_fk" FOREIGN KEY ("parent_id") REFERENCES "employees"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "employees_rels" ADD CONSTRAINT "employees_rels_parent_id_employees_id_fk" FOREIGN KEY ("parent_id") REFERENCES "employees"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "employees_rels" ADD CONSTRAINT "employees_rels_partners_id_partners_id_fk" FOREIGN KEY ("partners_id") REFERENCES "partners"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "partner_sites_rels" ADD CONSTRAINT "partner_sites_rels_parent_id_partner_sites_id_fk" FOREIGN KEY ("parent_id") REFERENCES "partner_sites"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "partner_sites_rels" ADD CONSTRAINT "partner_sites_rels_partners_id_partners_id_fk" FOREIGN KEY ("partners_id") REFERENCES "partners"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "partner_sites_rels" ADD CONSTRAINT "partner_sites_rels_deliverable_postal_codes_id_deliverable_postal_codes_id_fk" FOREIGN KEY ("deliverable_postal_codes_id") REFERENCES "deliverable_postal_codes"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;
`);

};

export async function down({ payload }: MigrateDownArgs): Promise<void> {
await payload.db.drizzle.execute(sql`

DO $$ BEGIN
 CREATE TYPE "enum_users_roles" AS ENUM('super-admin', 'admin');
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

ALTER TABLE "portals_rels" DROP CONSTRAINT "portals_rels_partner_sites_id_partner_sites_id_fk";

ALTER TABLE "payload_preferences_rels" DROP CONSTRAINT "payload_preferences_rels_employees_id_employees_id_fk";

ALTER TABLE "users_roles" DROP CONSTRAINT "employees_roles_parent_id_employees_id_fk";

ALTER TABLE "users_rels" DROP CONSTRAINT "employees_rels_parent_id_employees_id_fk";

ALTER TABLE "users_rels" DROP CONSTRAINT "employees_rels_partners_id_partners_id_fk";

ALTER TABLE "shipper_locations_rels" DROP CONSTRAINT "partner_sites_rels_parent_id_partner_sites_id_fk";

ALTER TABLE "shipper_locations_rels" DROP CONSTRAINT "partner_sites_rels_partners_id_partners_id_fk";

ALTER TABLE "shipper_locations_rels" DROP CONSTRAINT "partner_sites_rels_deliverable_postal_codes_id_deliverable_postal_codes_id_fk";

ALTER TABLE "employees_roles" RENAME TO "users_roles";
ALTER TABLE "employees" RENAME TO "users";
ALTER TABLE "employees_rels" RENAME TO "users_rels";
ALTER TABLE "partner_sites" RENAME TO "shipper_locations";
ALTER TABLE "partner_sites_rels" RENAME TO "shipper_locations_rels";
ALTER TABLE "portals_rels" RENAME COLUMN "partner_sites_id" TO "shipper_locations_id";
ALTER TABLE "payload_preferences_rels" RENAME COLUMN "employees_id" TO "users_id";
ALTER TABLE "users_roles" ALTER COLUMN "value" SET DATA TYPE enum_users_roles;
DO $$ BEGIN
 ALTER TABLE "portals_rels" ADD CONSTRAINT "portals_rels_shipper_locations_id_shipper_locations_id_fk" FOREIGN KEY ("shipper_locations_id") REFERENCES "shipper_locations"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "payload_preferences_rels" ADD CONSTRAINT "payload_preferences_rels_users_id_users_id_fk" FOREIGN KEY ("users_id") REFERENCES "users"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "users_roles" ADD CONSTRAINT "users_roles_parent_id_users_id_fk" FOREIGN KEY ("parent_id") REFERENCES "users"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "users_rels" ADD CONSTRAINT "users_rels_parent_id_users_id_fk" FOREIGN KEY ("parent_id") REFERENCES "users"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "users_rels" ADD CONSTRAINT "users_rels_partners_id_partners_id_fk" FOREIGN KEY ("partners_id") REFERENCES "partners"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "shipper_locations_rels" ADD CONSTRAINT "shipper_locations_rels_parent_id_shipper_locations_id_fk" FOREIGN KEY ("parent_id") REFERENCES "shipper_locations"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "shipper_locations_rels" ADD CONSTRAINT "shipper_locations_rels_partners_id_partners_id_fk" FOREIGN KEY ("partners_id") REFERENCES "partners"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;

DO $$ BEGIN
 ALTER TABLE "shipper_locations_rels" ADD CONSTRAINT "shipper_locations_rels_deliverable_postal_codes_id_deliverable_postal_codes_id_fk" FOREIGN KEY ("deliverable_postal_codes_id") REFERENCES "deliverable_postal_codes"("id") ON DELETE cascade ON UPDATE no action;
EXCEPTION
 WHEN duplicate_object THEN null;
END $$;
`);

};
```

### To Reproduce

1. Create one model (X) that has a relationship to another model (Y)
2. complete migrations to update the db
3. rename model X to something else
4. generate second migration
5. attempt to migrate

### Payload Version

2.4.0

### Adapters and Plugins

none

**Comment by jmikrut** [comment]
Hi there! This might be a Drizzle issue more than a Payload issue, but @DanRibbens we should update Drizzle to see if this can be resolved on its own with an update.

**Comment by stephaniesyed** [comment]
Thank you!

**Comment by stephaniesyed** [comment]
Should I report these directly to Drizzle?

**Comment by r1tsuu** [comment]
This is fixed on 3.0. On 2.0, it should be as well as soon as this PR will be released https://github.com/payloadcms/payload/pull/8468
